### PR TITLE
[rpc server]: Enable syncd-rpc target compilation.

### PR DESCRIPTION
**Pull request by oleksandrivantsiv**
Export switch ID variable required by SAI thrift.

### CODE REVIEW

File: syncd/syncd.cpp [hunk 1]
```diff
@@ -65,6 +65,13 @@ std::set<sai_object_id_t> initViewRemovedVidSet;
  */
 volatile bool g_asicInitViewMode = false;
 
+#ifdef SAITHRIFT
+/*
+ * SAI switch global needed for RPC server
+ */
+sai_object_id_t gSwitchId;
```
Comment (pavel-shirshov) [inline syncd/syncd.cpp]: What component does use the gSwitchId variable ?
I see you assign some value to it below, but how it is used?

Comment (oleksandrivantsiv) [reply syncd/syncd.cpp]: It is required by the latest SAI thrift implementation:
https://github.com/opencomputeproject/SAI/blob/master/test/saithrift/src/switch_sai_rpc_server.cpp#L80

Comment (pavel-shirshov) [reply syncd/syncd.cpp]: I'd better to make interface of SAI thrift implementation more explicit by introducing 
switch_id parameter to start_sai_thrift_rpc_server() function.
But anyway.

**Comment by lguohan** [comment]
why PR build fails? what do I need to add?

**Comment by andrii-savka** [comment]
We need to revert src/SAI to 1.0.2 version. The following commit can damage the build:
_commit aac4d6ecd6aef5d6a3f0805516bb3d515ec8291b
Author: Yakiv Huryk <yakivh@mellanox.com>
Date:   Sat Nov 18 12:35:41 2017 +0200

    PTF: alignment with 1.2 headers & fixes (#736)_

If I revert the src/SAI to following commit **_77b2ad28b8b6f3ad02105f209767dbda0fbba2a4_** then I have successfully built the image.

**Comment by lguohan** [comment]
retest this please

**Comment by andrii-savka** [comment]
@lguohan 
It was build and re-tested on the latest source code.

**Comment by andrii-savka** [comment]
@lguohan 
The build is failing because of syncd can't include "switch_sai_rpc_server.h" from SAI. 
I don't have such error in my local build. 
Probably, we need to update SAI. 
I'm investigating how to fix it.

**Comment by andrii-savka** [comment]
@lguohan 
The flow to build syncd with SAI thrift is the same. 
Could you check whether SAI which is used by Jenkins has a "switch_sai_rpc_server.h" thrift headers?

**Comment by pavel-shirshov** [comment]
This build requires libsaithrift-dev. This library build was broken when we migrate from 0.9.4 to 1.0.2.
I'll try to fix the whole build process in the buildimage repo first.
I'll write results here as soon as I found anything.

**Comment by pavel-shirshov** [comment]
Hi Guohan
You need to install libsaithrift-dev_1.0.4_amd64.deb
It should help you.

**Comment by andrii-savka** [comment]
Tested and created rebased PR: #281

**Comment by lguohan** [comment]
merged the #281

**Pull request closed without merging**